src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,960 @@
+
+        self.runtime.execution_config.operation = Constants.INSTALLATION
+        self.runtime.status_handler.set_current_operation(Constants.INSTALLATION)
+
+        patch_count_pending = random.randint(1, 400)
+        patch_count_exclude = random.randint(1, 100)
+        patch_count_not_selected = random.randint(780, 1000)
+
+        # random_char=random.choice(string.ascii_letters) ensure the packages are unique
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count_pending)
+        self.runtime.status_handler.set_package_install_status(test_packages, test_package_versions)
+
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count_exclude, random_char=random.choice(string.ascii_letters))
+        self.runtime.status_handler.set_package_install_status(test_packages, test_package_versions, Constants.EXCLUDED)
+
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count_not_selected, random_char=random.choice(string.ascii_letters))
+        self.runtime.status_handler.set_package_install_status(test_packages, test_package_versions, Constants.NOT_SELECTED)
+
+        self.runtime.status_handler.set_installation_substatus_json(status=Constants.STATUS_SUCCESS)
+
+        patch_count = patch_count_pending + patch_count_exclude + patch_count_not_selected
+
+        # Assert complete status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.complete_status_file_path, 'r') as file_handle:
+            complete_substatus_file_data = json.load(file_handle)
+
+        installation_msg = self.__get_message_json_from_substatus(complete_substatus_file_data)
+
+        # Assert complete status file
+        self.__assert_patch_summary_from_status(complete_substatus_file_data, Constants.INSTALLATION, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_SUCCESS, patch_count,
+            installation_msg, under_internal_size_limit=False)
+
+        # Assert installation truncated status file
+        with self.runtime.env_layer.file_system.open(self.runtime.execution_config.status_file_path, 'r') as file_handle:
+            truncated_substatus_file_data = json.load(file_handle)
+
+        installation_truncated_msg = self.__get_message_json_from_substatus(truncated_substatus_file_data)
+        # Assert truncated status file size < completed status file size
+        self.assertTrue(len(json.dumps(truncated_substatus_file_data)) < len(json.dumps(complete_substatus_file_data)))
+        self.__assert_patch_summary_from_status(truncated_substatus_file_data, Constants.INSTALLATION, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_WARNING, patch_count, installation_truncated_msg)
+
+        # Assert all installation fields in the message json are equal in both status files
+        self.__assert_installation_truncated_msg_fields(installation_msg, installation_truncated_msg)
```

Comment:
How is this test ensuring only the low pri packages were removed? There's no assertions validating it

Reply:
i've added assertion to check last patch state is not_selected prior truncation, post truncation check first and last patch state is not not_selected
done